pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -56,23 +57,70 @@ func GetLocalityLbSetting(
 	return mesh
 }
 
-func ApplyLocalityLBSetting(
-	locality *core.Locality,
+func GetNetworkLbSetting(
+	mesh *v1alpha3.NetworkLoadBalancerSetting,
+	destrule *v1alpha3.NetworkLoadBalancerSetting,
+) *v1alpha3.NetworkLoadBalancerSetting {
+	var enabled bool
+	// Locality lb is enabled if its not explicitly disabled in mesh global config
+	if mesh != nil && (mesh.Enabled == nil || mesh.Enabled.Value) {
```

Comment:
nit: I think you can assign enabled = mesh.Enabled.Value same for Drs below. Code might be simplified